### PR TITLE
fix: donot set l0 segment as growing when savebinlogs

### PR DESCRIPTION
**Pull request by XuanYang-cn**
This PR fixes negative growing L0 segments in Metrics

See also: #29204, #30441

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
#29204

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/datacoord/meta.go [hunk 1]
```diff
@@ -229,17 +229,20 @@ func (m *meta) GetSegmentsChanPart(selector SegmentInfoSelector) []*chanPartSegm
 		if !selector(segmentInfo) {
 			continue
 		}
-		dim := fmt.Sprintf("%d-%s", segmentInfo.PartitionID, segmentInfo.InsertChannel)
+
+		cloned := segmentInfo.Clone()
```
Comment (czs007) [inline internal/datacoord/meta.go:233]: why clone this?

**Pull request merged**